tests/jobs/scala/src/main/scala/KerberizedKafkaProducer.scala

```diff
@@ -35,14 +35,28 @@ object KerberizedKafkaProducer {
 
   def sendMessages(topic: String, messages: Array[String])(producer: KafkaProducer[Int, String]): Unit = {
     messages.foreach { str =>
-      val msg = new ProducerRecord[Int, String](topic, str.hashCode, str)
+      val msg = new ProducerRecord[Int, String](topic, str)
       println(s"sending message: $msg")
 
-      Try(producer.send(msg)) match {
-        case Success(_) => println("Message sent")
-        case Failure(ex) => ex.printStackTrace()
+      producer.send(msg, new Callback {
+      override def onCompletion(metadata: RecordMetadata, exception: Exception): Unit = {
+        if (exception != null) {
+          exception.printStackTrace()
+        } else {
+          println(s"Message has been published to (topic_partition@offset): ${metadata.toString}")
```

Comment:
is the `metadata` always present when there's no exception?

Reply:
Yes, it can be `null` only when an error is occurred